src/FSharp.Data.GraphQL.Server/Values.fs

```diff
@@ -89,10 +89,35 @@ let rec internal compileByType (inputObjectPath: FieldPath) (inputSource : Input
 
     | InputObject objDef ->
         let objtype = objDef.Type
-        let ctor = ReflectionHelper.matchConstructor objtype (objDef.Fields |> Array.map (fun x -> x.Name))
+        let (constructor : obj[] -> obj), (parameterInfos : Reflection.ParameterInfo[]) =
+            if typeof<IDictionary<string,obj>>.IsAssignableFrom(objtype) then
+                let parameterInfos = [|
+                    for f in objDef.Fields ->
+                        { new Reflection.ParameterInfo() with
+                            member _.Name = f.Name
+                            member _.ParameterType = f.TypeDef.Type
+                            member _.Attributes =
+                                match f.TypeDef with
+                                | Nullable _ -> Reflection.ParameterAttributes.Optional
+                                | _ -> Reflection.ParameterAttributes.None
+                        }
+                |]
+                let constructor (args:obj[]) =
+                    let o = Activator.CreateInstance(objtype)
+                    let dict = o :?> IDictionary<string, obj>
+                    for fld,arg in Seq.zip objDef.Fields args do
+                        match arg, fld.TypeDef with
+                        | null, Nullable _ -> () // skip populating Nullable fields with nulls
```

Comment:
Why do you want this to happen? If null came and if no filed passed at all are different cases